c2rust-refactor/src/scripting.rs

```diff
@@ -587,5 +838,64 @@ impl<'a, 'tcx> UserData for TransformCtxt<'a, 'tcx> {
         methods.add_method("get_ast", |lua_ctx, this, node: LuaAstNode| {
             this.get_lua_ast(lua_ctx, node)
         });
+
+        /// Visits all function like items
+        // @function visit_fn_like
+        // @tparam LuaAstNode krate Crate in its entirety
+        // @tparam function() callback Function called for each function like item.
+        methods.add_method_mut("visit_fn_like", |lua_ctx, this, (krate, callback): (LuaAstNode, LuaFunction)| {
+            let mut krate = ast::Crate::try_from(this.remove_ast(krate)).expect("Did not find crate input");
+            let mut found_err = Ok(());
+
+            mut_visit_fns(&mut krate, |fn_like| {
```

Comment:
I don't like that the names are different: `visit_fn_like` in Lua and `visit_fns`/`mut_visit_fns` in Rust. Ideally, the names and perhaps signatures would be identical between the two languages, if the Lua version is just a thin wrapper over the Rust one.

Reply:
I think this is getting moved to the `Visitor` interface, right @TheDan64? If so, it will be a method that isn't part of the libsyntax `Visitor` trait, since the concept of `fn_like`s is something we have for the refactorer. I think the name is fine if it is moved.

Reply:
That's the general idea, yes. But I don't think I'll be able to use the existing visitor trait as I need to walk the Lua table AST rather than the rust AST